web/src/components/InfiniteScrollList.tsx

```diff
@@ -62,14 +63,14 @@ const InfiniteScrollList = <T,>({
   }, [page, hasMore, itemsPerPage, loadPage])
 
   useEffect(() => {
-    if (resetOnLanguageChange) {
+    if (resetOnLanguageChange || filterValues) {
       setData([])
       setError(null)
       setLoading(false)
       setHasMore(true)
       setPage(defaultPage)
     }
-  }, [languageCode, resetOnLanguageChange, defaultPage])
+  }, [languageCode, resetOnLanguageChange, defaultPage, filterValues])
```

Comment:
Does this work? I think we could just use the dependency on `[loadData]`, that function should change on language change/filter value change already.

Reply:
you have to reset the data first and then the load gets triggered auto (on filter changed)

Reply:
see suggestion below